src/components/BillFilter/__tests__/BillFilter.test.jsx

```diff
@@ -1,134 +1,198 @@
+        dateOpened: false,
       });
-      it("BillFilter should match snap shot if category filter is active", () => {
-        wrapper.setProps({
-          filterToggles: {
-            ...SAMPLE_FILTER_TOGGLES,
-            categoryOpened: true
-          }
-        })
-        matches(wrapper);
+      wrapper.find("Badge").at(6).simulate("click");
+      expect(mockFilterToggleChange).toBeCalledWith({
+        dateOpened: true,
+        categoryOpened: false,
+        statusOpened: false,
       });
+      expect(mockFilterToggleChange).toBeCalledTimes(3);
     });
-  });
 
-  describe("component ", () => {
     it("should show the date value next to start date", () => {
-      wrapper.setProps({
-        dateFilters:{
-          startDate: { selected: true, value: "2020-01-01" },
-          endDate: { selected: false, value: "" },
-        },
-        filterToggles: {
-          statusOpened: false,
-          categoryOpened: false,
-          dateOpened: true,
-        }
-      });
+      const dateFilters = {
+        startDate: { selected: true, value: "2020-01-01" },
+        endDate: { selected: false, value: "" },
+      };
+      const filterToggles = {
+        statusOpened: false,
+        categoryOpened: false,
+        dateOpened: true,
+      };
+
+      wrapper.setProps({ dateFilters, filterToggles });
 
       expect(wrapper.find("FormCheckbox").at(0).render().text()).toBe(
         "Start date: 2020-01-01"
       );
     });
 
     it("should show the date value next to end date", () => {
-      wrapper.setProps({
-        dateFilters:{
-          startDate: { selected: false, value: "" },
-          endDate: { selected: true, value: "2020-01-01" },
-        },
-        filterToggles: {
-          statusOpened: false,
-          categoryOpened: false,
-          dateOpened: true,
-        }
-      });
+      const dateFilters = {
+        startDate: { selected: false, value: "" },
+        endDate: { selected: true, value: "2020-01-01" },
+      };
+      const filterToggles = {
+        statusOpened: false,
+        categoryOpened: false,
+        dateOpened: true,
+      };
+      wrapper.setProps({ dateFilters, filterToggles });
 
       expect(wrapper.find("FormCheckbox").at(1).render().text()).toBe(
         "End date: 2020-01-01"
       );
     });
+
+    it("should trigger mockDateCheckboxHandler and handleDateSelection", () => {
+      const filterToggles = {
+        statusOpened: false,
+        categoryOpened: false,
+        dateOpened: true,
+      };
+      wrapper.setProps({ filterToggles });
+
+      wrapper.find("FormCheckbox").at(0).simulate("change");
+      expect(mockDateCheckboxHandler).toBeCalledWith("startDate");
+      wrapper.find("FormCheckbox").at(1).simulate("change");
+      expect(mockDateCheckboxHandler).toBeCalledWith("endDate");
+      const event = {
+        target: { value: "2020/01/04" },
+      };
+      wrapper.find("FormInput").at(0).simulate("change", event.target.value);
```

Comment:
shouldn't it be:

```suggestion
      wrapper.find("FormInput").at(0).simulate("change", event);
```

?

Reply:
and if it passed like this, then it goes to show the test is not really proving much here...